helix-view/src/view.rs

```diff
@@ -161,6 +161,62 @@ impl View {
         Some(Position::new(row, col))
     }
 
+    /// Verifies whether a screen position is inside the view
+    /// Returns true when position is inside the view
+    pub fn verify_screen_coords(&self, row: usize, column: usize) -> bool {
+        // 2 for status
+        if row < self.area.y as usize || row > self.area.y as usize + self.area.height as usize - 2
+        {
+            return false;
+        }
+
+        // TODO: not ideal
+        const OFFSET: usize = 7; // 1 diagnostic + 5 linenr + 1 gutter
+
+        if column < self.area.x as usize + OFFSET
+            || column > self.area.x as usize + self.area.width as usize
+        {
+            return false;
+        }
+        true
+    }
+
+    /// Translates a screen position to position in the text document.
+    /// Returns a usize typed position in bounds of the text if found in this view, None if out of view.
+    pub fn pos_at_screen_coords(&self, doc: &Document, row: usize, column: usize) -> Option<usize> {
```

Comment:
Instead of returning `None` when the cursor is outside of the text, it might be more useful to return the "closest" valid position in the text.  For example, even if the mouse is off the end of a line, return the position of the last character on that line.  This would make it easier to e.g. quickly drag-select a bunch of lines without having to be too precise.

I suspect that's also the behavior we would want if we eventually use this function elsewhere, too.

Reply:
Also, we'll definitely want unit tests on this.  It might be worth breaking out a sub-function that does all of the actual logic, but takes a `RopeSlice` instead of a `Document`, so that unit tests are easier to write.